System/Matrix.cpp

```diff
@@ -38,8 +38,8 @@ namespace RTE {
 		m_ElementsUpdated = true;
 
 		// Inverse angle to make CCW positive direction.
-		float const CosAngle = static_cast<float>(std::cos(-angle));
-		float const SinAngle = static_cast<float>(std::sin(-angle));
+		const float CosAngle = std::cos(-angle);
```

Comment:
These should `std::sinf` and `std::cosf`.
Not entirely sure how they're different but since we're using the `f` ones for other stuff it's better to change for consistency.
Same in `UpdateElements()` (line 178-179).

Reply:
I kept it like this so the intrinsic functions may be used, as we have this on the top of the file:
`#pragma intrinsic (sin, cos)`
For sinf and cosf this wouldn't work, I think. Right?

Reply:
Hmmm fair point, I don't know.
I don't even know if it works as is because CRT `sin` and `std::sin` might be different.
And I don't know of a way to test it out either.

But i'm guessing that regardless of what is being used at the long chain of overloads and overrides and the clusterfuck of cpp regular C `sin` is being used. Same for everything else.

So yeah, don't know, let's change them and pray.

Reply:
ok, looks like sinf,cosf will work as well. moreover, the pragma declaration is useless since we already have, in "Final", the compiler option -Oi which means that whenever the compiler thinks that it's good, then it replaces functions with an available intrinsic version.
[link1](https://docs.microsoft.com/en-us/cpp/build/reference/oi-generate-intrinsic-functions?view=vs-2019)
[link2](https://docs.microsoft.com/en-us/cpp/intrinsics/intrinsics-available-on-all-architectures?view=vs-2019)

Reply:
Alright great, so just remove the pragma cause there's no point keeping it there and we're done here.